src/workerd/api/container.c++

```diff
@@ -65,7 +65,6 @@ jsg::Promise<void> Container::monitor(jsg::Lock& js) {
       js.throwException(kj::mv(error));
     }
   }, [this](jsg::Lock& js, jsg::Value&& error) {
-    running = false;
```

Comment:
This will change behavior even in prod, are we aligned with CC team?

Reply:
cc @mikenomitch

Reply:
When monitor fails I think we should keep it that running should be false _for now_  as monitor might return an error if there is an exit code that is not zero, or other internal issues. I think we want Monitor() to not throw an error and return an exit code, and make workerd throw an error with an exitCode property:
https://github.com/cloudflare/workerd/pull/4369#discussion_r2155076343